sn_networking/src/record_store.rs

```diff
@@ -1461,120 +1460,140 @@ mod tests {
         Ok(())
     }
 
+    struct PeerStats {
+        address: NetworkAddress,
+        records_stored: usize,
+        nanos_earned: u64,
+        payments_received: usize,
+    }
+
     #[test]
     fn address_distribution_sim() {
-        // Map of peers and correspondent stats of `(num_of_records, Nano_earned, received_payment_count)`.
-        let mut peers: HashMap<PeerId, (usize, u64, usize)> = Default::default();
-        let mut peers_vec = vec![];
-
+        use rayon::prelude::*;
         let num_of_peers = 2000;
         let num_of_chunks_per_itr = 2000;
-
-        for _ in 0..num_of_peers {
-            let peer_id = PeerId::random();
-            let _ = peers.insert(peer_id, (0, 0, 0));
-            peers_vec.push(peer_id);
-        }
+        let close_group_size = close_group_majority().min(num_of_peers);
```

Comment:
I think this shall be replication_range to be used.
in practical, this shall be a kbucket_index
in the sim, we used REPLICATION_PEERS_COUNT (7) to simulate, which shall actually be updated to (K_VALUE/2) to be more close to the real situation.
but change to majority of close_group_size (which is just 3) will be too aggressive